lib/adyen/rest.rb

```diff
@@ -26,11 +26,7 @@ module REST
     # @return [Adyen::REST::Client] A configured client instance
     # @see .session
     def self.client
-      Adyen::REST::Client.new(
-        Adyen.configuration.environment,
-        Adyen.configuration.api_username,
-        Adyen.configuration.api_password
-      )
+      Adyen::REST::Client.new(Adyen.configuration.environment)
```

Comment:
Ah, OK, so it's just that you have multiple accounts with different permissions. In that case, my suggestion would be to create separate clients for them.

``` ruby
api_client = Adyen::Client.new(env, api_username, api_password)
payout_client = Adyen::Client.new(env, payout_username, payout_client)
```

This allows us to get rid of the ugly logic in the initializer. Note that this will only work of Client never talks to `Adyen.configuration` directly.